proxy/tests/support/proxy.rs

```diff
@@ -61,6 +65,16 @@ impl Proxy {
         self
     }
 
+    pub fn disable_inbound_ports_protocol_detection(mut self, ports: Vec<u16>) -> Self {
+        self.inbound_disable_ports_protocol_detection = Some(ports);
+        self
+    }
```

Comment:
Sorry i didn't notice this before. I think it would be better to just have the tests set `ENV_INBOUND_PORTS_DISABLE_PROTOCOL_DETECTION` directly, instead of offering this convenience API for them. That would more closely match what happens in the production code.